src/libmongoc/doc/learn/get/from-source.rst

```diff
@@ -0,0 +1,386 @@
+
+__ https://github.com/mongodb/mongo-c-driver
+
+.. important::
+
+  It is **highly recommended** that new users use a stable released version of
+  the driver, rather than building from a development branch. When you
+  `git clone` or download an archive of the repository, be sure to specify a
+  release tag (i.e. with Git's `--branch` argument).
+
+.. note::
+
+  In the following examples, `$SOURCE` refers to the directory in which you will
+  be placing the |mongo-c-driver| source code, and `$VERSION` refers to the
+  version of the C driver that you are building. The current version written for
+  this documentation is |version.pre|.
+
+.. seealso::
+  For the list of supported platforms, refer to the :doc:`/ref/platforms` page.
+
+Obtaining the Source
+********************
+
+There are two primary recommended methods of obtaining the |mongo-c-driver|
+source code:
+
+1. Clone the repository using `git`. :ref:`(See below) <learn.obtaining.git>`
+
+   - This is useful if you wish to view the version history, test unreleased
+     features/fixes, or contribute to the codebase.
+   - This is straightforward and an equivalent process across all platforms.
+
+2. Download a source archive at a specific version.
+   :ref:`(See below) <learn.obtaining.archive>`
+
+   - This is useful without needing to have `git` available and can be done
+     using only built-in programs that ship with most major operating systems,
+     but requires a slightly different process between platforms.
+   - The download is smaller and stored in a single file.
+
+
+.. _learn.obtaining.git:
+
+Downloading Using Git
+=====================
+
+Using Git, the C driver repository can be cloned from the GitHub URL
+``https://github.com/mongodb/mongo-c-driver.git``. Git tags for released
+versions are named after the version for which they correspond (e.g.
+"|version.pre|"). To clone the repository using the command line, the following
+command may be used::
+
+  $ git clone https://github.com/mongodb/mongo-c-driver.git --branch="$VERSION" "$SOURCE"
+
+.. hint::
+
+  Despite the name, `git-clone`\ 's `--branch` argument may also be used to
+  clone from repository *tags*.
+
+.. _learn.obtaining.archive:
+
+Downloading a Release Archive
+=============================
+
+An archived snapshot of the repository can be obtained from the
+`GitHub Releases Page`__. The ``mongo-c-driver-x.y.z.tar.gz`` archive attached
+to any release contains a minimal version of the repository with only the files
+necessary to build the driver binaries.
+
+__ https://github.com/mongodb/mongo-c-driver/releases
+
+The archive can be downloaded and extracted from a command line using one of
+several tools, depending on what is available.
+
+.. tab-set::
+
+  .. tab-item:: Using ``wget`` + ``tar``
+
+    ::
+
+      ## Download using wget:
+      $ wget "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/$VERSION.tar.gz" \
+          --output-document="mongo-c-driver-$VERSION.tar.gz"
```

Comment:
The source archive from GitHub does not include the `VERSION_CURRENT` file. 

Running the following:
```sh
VERSION=1.24.3                      
wget "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/$VERSION.tar.gz" \
    --output-document="mongo-c-driver-$VERSION.tar.gz"
tar xf mongo-c-driver-$VERSION.tar.gz
cd mongo-c-driver-$VERSION
cmake -S=. -B=cmake-build \
    -D ENABLE_EXTRA_ALIGNMENT=FALSE \
    -D ENABLE_AUTOMATIC_INIT_AND_CLEANUP=FALSE \
    -D CMAKE_BUILD_TYPE=RelWithDebInfo \
    -D ENABLE_MONGOC=OFF
```

Results in this output from CMake:
```
-- calculated BUILD_VERSION 0.0.0
```

This may result in consumers getting an error when attempting to use the build with version 0.0.0.
Consider adding `-D BUILD_VERSION=$VERSION` to the CMake step.